### ISSUE TITLE
Stress/Performance Testing

### ISSUE THREAD

**Issue by pivnicek** [keboola/db-extractor-common#32]
It would be great to have a test that you can run to test performance versus the previous version of the component.

pseudocode:
```
oldVerTestStart = now()
run [theCurrentLiveImage] stressTest
oldImageDuration = now() - oldVerTestStart

newVerTestStart = now()
run [theLocalImage] stressTest
newImageDuration = now() - newVerTestStart

if newImageDuration / oldImageDuration > 1 // or some env ratio threshold
    throw tantrum
```

**Comment by Halama** [comment]
Maybe it can just store the results somewhere (S3?) and than download it after new execution and just compare it or show some histogram. And at the end save new results. The inspiration is from zonky meetup and performance tests https://www.facebook.com/events/173333633393912/

**Comment by pivnicek** [comment]
yeah, we can store the results to s3 for sure, but I like the idea of being able to fail the build on huge performance degradation.

**Comment by Halama** [comment]
yes, that was the idea. You will download it and compare last run and if there will be degradation you'll fail the build. And on success run you'll just add a new record of current run.

**Comment by pivnicek** [comment]
Ah, I see.  yes, that would require only needing to run the test once.  cool, sounds good.
